server/src/main/java/org/elasticsearch/inference/InferenceService.java

```diff
@@ -52,7 +52,20 @@ public interface InferenceService extends Closeable {
      * @param secrets Sensitive configuration options (e.g. api key)
      * @return The parsed {@link Model}
      */
-    Model parsePersistedConfig(String modelId, TaskType taskType, Map<String, Object> config, Map<String, Object> secrets);
+    Model parsePersistedConfigWithSecrets(String modelId, TaskType taskType, Map<String, Object> config, Map<String, Object> secrets);
+
+    /**
+     * Parse model configuration from {@code config map} from persisted storage and return the parsed {@link Model}.
+     * This function modifies {@code config map}, fields are removed from the map as they are read.
+     *
+     * If the map contains unrecognized configuration options, no error is thrown.
+     *
+     * @param modelId Model Id
+     * @param taskType The model task type
+     * @param config Configuration options
+     * @return The parsed {@link Model}
+     */
+    Model parsePersistedConfig(String modelId, TaskType taskType, Map<String, Object> config);
```

Comment:
I don't understand why we need this with/without secrets for this feature.

Reply:
Secrets shouldn't be returned as part of the get request when a user simply retrieves a model. They're used when an inference request is made to a 3rd party service. The model configuration is stored in elasticsearch and parsed and converted to the response format. The parsing logic is specific to each service type (openai, elser, hugging face etc).

I think the reason we want a way to parse the model without worrying about secrets is because to retrieve the secrets we have to get them from a separate index so it makes the retrieval process more complicated when handling multiple models.

To retrieve a single model we do a bulk get to retrieve the secrets and configurations in one go.

`parsePersistedConfig` allows the parsing logic to ignore the fact that the secrets weren't retrieved, otherwise we'd throw an error while parsing. I suppose we could pass a boolean that indicates whether we should ignore parsing the secrets.